monitoring/rid_qualifier/aircraft_state_replayer.py

```diff
@@ -0,0 +1,146 @@
+            test_flight_details = TestFlightDetails(effective_after = test_start_offset_isoformat,details = [rid_flight_details])
+            test_flight = TestFlight(injection_id = str(uuid.uuid4()), telemetry = updated_timestamps_telemetry, details_responses = test_flight_details)
+            test_flight_deserialized = self.make_json_compatible(test_flight)
+            requested_flights.append(test_flight_deserialized)
+            test_payload = TestPayload(test_id = test_id, requested_flights = requested_flights)
+            test_payload_data_metadata = DeliverablePayloads(injection_path = test_injection_path, injection_payloads = test_payload)
+            all_test_payloads.append(test_payload_data_metadata)
+        
+        return all_test_payloads
+
+
+class TestHarness():
+    ''' A class to submit Aircraft RID State to the USS test endpoint '''
+
+    def __init__(self, auth_spec:str, injection_base_url:str):
+        
+        auth_adapter = make_auth_adapter(auth_spec)
+        self.uss_session = DSSTestSession(injection_base_url, auth_adapter)
+
+    def submit_test(self,uss_session:DSSTestSession, payload:DeliverablePayloads) -> None:
+        
+        response = uss_session.put(url = payload.injection_path, json=payload.injection_payloads, scope = ' '.join([rid.SCOPE_RID_QUALIFIER_INJECT ]))
+
+        if response.status_code == 200:
+            print("New test with ID %s created" % payload['injection_payload']['test_id'])
+        elif response.status_code ==409:
+            raise RuntimeError("Test with ID %s already exists" % payload['injection_payload']['test_id'])
+        elif response.status_code == 404:
+            raise RuntimeError("Test with ID %s not submitted, the requested endpoint was not found on the server" % payload['injection_payload']['test_id'])
+        elif response.status_code == 401:
+            raise RuntimeError("Test with ID %s not submitted, the access token was not provided in the Authorization header, or the token could not be decoded, or token was invalid" % payload['injection_payload']['test_id'])
+        elif response.status_code == 403:
+            raise RuntimeError("Test with ID %s not submitted, the access token was decoded successfully but did not include the appropriate scope" % payload['injection_payload']['test_id'])
+        elif response.status_code == 413:
+            raise RuntimeError("Test with ID %s not submitted, the injection payload was too large" % payload['injection_payload']['test_id'])
+        else:
+            raise RuntimeError("Test with ID %(test_id)s not submitted, the server returned the following HTTP error code: %(status_code)d" % {'test_id':payload['injection_payload']['test_id'], 'status_code': response.status_code})
```

Comment:
Because `payload` is a typed `DeliverablePayloads`, we can see what keys/properties are available in the definition -- `injection_payload` is not one of those available keys/properties.  This can be changed to:

```python
'test_id': ', '.join(p.test_id for p in payload.injection_payloads)
```

Reply:
I have now created a new test_id variable to access the id during the `put` call.